src/main.rs

```diff
@@ -75,33 +93,87 @@ fn idle(hosts: &[String]) -> MPDClient {
     }
 }
 
+fn format_duration(duration: Option<&String>) -> Option<String> {
+    if duration.is_some() {
+        let time = duration.unwrap().parse::<f32>().unwrap();
+
+        let seconds = (time % 60.0).round();
+        let minutes = ((time % 3600.0) / 60.0).round();
+
+        let formatted = format!("{:0>2}:{:0>2}", minutes, seconds);
+        return Some(formatted);
+    }
+
+    None
+}
+
+fn get_token_value(song: &Song, token: &str) -> String {
+    match token {
+        "title" => song.title.as_ref().unwrap_or(&EMPTY).clone(),
+        "album" => song.tags.get("Album").unwrap_or(&EMPTY).clone(),
+        "artist" => song.tags.get("Artist").unwrap_or(&EMPTY).clone(),
+        "date" => song.tags.get("Date").unwrap_or(&EMPTY).clone(),
+        "disc" => song.tags.get("Disc").unwrap_or(&EMPTY).clone(),
+        "genre" => song.tags.get("Genre").unwrap_or(&EMPTY).clone(),
+        "track" => song.tags.get("Track").unwrap_or(&EMPTY).clone(),
+        "duration" => format_duration(song.tags.get("duration")).unwrap_or(String::new()),
+        _ => token.to_string(),
+    }
+}
```

Comment:
I want to re-write this to be less disgusting before merging it

Reply:
If it's any consolation, I don't think it's disgusting.

Granted, I am by no means a professional programmer. Everything I've ever done has been self-taught, so I'm sure the way I see things is much different than someone who is a professional.

Reply:
It's clear what it does, but it's a lot of repeated code and it's certainly not a very efficient approach. This is my first real program in Rust so I'm still trying to get comfortable with the basics.

If I don't get anywhere in the next couple of days, I'll merge it as-is and re-write it some other day

Reply:
Many thanks to @simvux for tidying this up for me!